### PR TITLE
feat*: remove the need for symlinks by using two variables

### PR DESCRIPTION
**Pull request by wdconinc**
We pretty much require setting environment variables already, so this just formalizes this. Based on #23 and discussion therein.

For consistency, ip6 should also install its xml files into `share/ip6/compact` but it doesn't exactly because we used a symlink and would conflict with the existing `compact` directory of the detector repository.

### CODE REVIEW

File: templates/epic.xml.jinja2 [hunk 1]
```diff
@@ -44,15 +44,15 @@
...
+    <include ref="${DETECTOR_PATH}/compact/definitions.xml" />
   </define>
 
   <includes>
-    <gdmlFile ref="compact/elements.xml"/>
-    <gdmlFile ref="compact/materials.xml"/>
-    <file     ref="compact/optical_materials.xml"/>
+    <gdmlFile ref="${DETECTOR_PATH}/compact/elements.xml"/>
+    <gdmlFile ref="${DETECTOR_PATH}/compact/materials.xml"/>
+    <file     ref="${DETECTOR_PATH}/compact/optical_materials.xml"/>
```
Comment (veprbl) [inline templates/epic.xml.jinja2:55]: I have the same reservation about using DETECTOR_PATH as in https://github.com/eic/ip6/pull/9#issuecomment-1200526270. Would be interested to hear your line of thinking when you made this.

Comment (veprbl) [reply templates/epic.xml.jinja2:55]: I just had to add those `${DETECTOR_PATH}` when creating my own composition for the EPIC detector by first copying an existing xml from a package to my local directory. Having this PR as is would be useful for that case.

**Review by veprbl — APPROVED** [review]

**Comment by wdconinc** [comment]
Reasons I have put this on hold is that I realized that we use a mixture of `BEAMLINE` and `BEAMLINE_CONFIG` variables, followed by `VERSION` etc in a way that is not consistent with how we treat `BEAMLINE`.

Now I have verified that at least 1) we are consistent in the definition and use of `DETECTOR_PATH`, but 2) we don't define `BEAMLINE_PATH` yet in the containers. So, it depends on rolling this into [eic_container](https://eicweb.phy.anl.gov/containers/eic_container) and onto cvmfs.

**tl;dr** I'll merge this tomorrow.

**Comment by wdconinc** [comment]
The `setup.sh` in the geometry repos has it right; it's the `setup.sh` generated by [setup_detectors.py](https://eicweb.phy.anl.gov/containers/eic_container/-/blob/master/containers/jug/setup_detectors.py) that doesn't yet.

**Comment by wdconinc** [comment]
Confirmed that `${BEAMLINE_PATH}` (and ${DETECTOR_PATH}`) is now set in the `eic-shell` containers and that `checkGeometry -c epic_sciglass.xml` now works without errors inside `eic-shell` after loading the correct `/opt/detector/epic-nightly/setup.sh`.

Potentially failure scenario to be mindful of:
1. Users loads `/opt/detector/setup.sh` and expects `checkGeometry -c epic_sciglass.xml` to work, but `/opt/detector/setup.sh` points to a non-EPIC geometry (as it does not)...

Nevertheless, ready for merging.
